File: lab3/FileReverse.c
/*-------------------------------------------------------------------------
 *Amanda Nguyen 
 *amvanguy@ucsc.edu
 *CMPS 12M: Tantalo (Summer 2017)
 *
 *Lab #3: The purpose of this assignment is to  introduce the C programming 
 *language, including standard input-output functions, command line arguments, 
 *File IO, and compilation with Makefiles.
 *------------------------------------------------------------------------
 * FileReverse.c
 * Take two command line arguments naming the input and output files
 * respectively, read each word in the input file, then
 * print it backwards on a line by itself.
 *------------------------------------------------------------------------
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/*reverse function, swaps letters within word until reversed*/
void stringReverse (char *s){

	if(s == 0 || *s == 0){
		return;
	}

	char *i = s;
    char *j = i + strlen(s) - 1;
    char temp; 

    while(i < j){
    	temp = *i;
    	*i = *j;
    	*j = temp;

    	i++;
    	j--;
 	}
}

int main(int argc, char* argv[]){

	/*declarations of variables*/
	FILE* in;   
	FILE* out; 
	char word[256]; /*allocation of 256 bits for input file*/
	size_t strlen(const char *str);

   /* check command line for correct number of arguments */
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   /* open input file for reading */
   in = fopen(argv[1], "r");

   if(in == NULL){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   /* open output file for writing */
   out = fopen(argv[2], "w");

   if(out == NULL){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   /* read words from input file, print on separate lines to output file*/
   while( fscanf(in, " %s", word) != EOF){
   		stringReverse(word); 
    	fprintf(out, "%s\n", word);
   }

   /* close input and output files */
   fclose(in);
   fclose(out);

   return(EXIT_SUCCESS);
}

File: lab2/FileReverse.java
//-------------------------------------------------------------------------
//Amanda Nguyen 
//amvanguy@ucsc.edu
//CMPS 12M: Tantalo (Summer 2017)
//
//Lab #2: The purpose of this assignment is to practice using command line 
//arguments, file input-output, and manipulation of Strings in java
//-------------------------------------------------------------------------
// FileReverse.java 
// Takes two command line arguments giving the names of the input and output files respectively
// ------------------------------------------------------------------------

import java.io.*;
import java.util.Scanner;
import java.io.FileNotFoundException;

class FileReverse{

	public static void main(String[] args) throws IOException{

 		Scanner in = null;
 		PrintWriter out = null;
 		String line = null;
 		String[] token = null; 
 		int lineCount = 0;

 		// check number of command line arguments is at least 2, print usage message
 		if(args.length < 2){
 			System.out.println("Usage: FileCopy <input file> <output file>");
 			System.exit(1);
 		}
 
 		//first argument = input file, second argument, output file
 		in = new Scanner(new File(args[0]));
 		out = new PrintWriter(new FileWriter(args[1]));
 
 		// read lines from in, write lines to out
 		while(in.hasNextLine()){	//if file has next line
 			lineCount++; 
	         
	        // trim leading and trailing spaces of orig. line, 
	        // then add one trailing space so split works on blank lines

	        line = in.nextLine().trim() + " "; 


	        //split line around white space, puts each token into array 
	        token = line.split("\\s+");

	        for(int i = 0; i < token.length; i++){
	        	String reversed = stringReverse(token[i], token[i].length());
	        	out.println(reversed);
	        	// token[i] = reversed;
	        	// out.println(token[i]);
	        }
 		}
 
 		System.out.println(lineCount);
 		//close files
 		in.close();
 		out.close();
 	}

 	//'stringReverse' function reverses strings using recursion 
 	public static String stringReverse(String s, int n){

 		if(n > 0){
 			return (s.charAt(n-1) + stringReverse(s, n-1));
 		}else{
 			return " ";
 		}
 	}
}